charts/tests/ut/cp_network-attachments.bats

```diff
@@ -0,0 +1,123 @@
+    template --set-json 'interfaces=[{"type": "multus"}]'
+    assert_query_equal '.metadata.namespace' "default"
+}
+
+@test "Control Plane NetworkAttachmentDefinition: No annotations are set by default" {
+    template --set-json 'interfaces=[{"type": "multus"}]'
+    assert_query '.metadata.annotations | not'
+}
+
+@test "Control Plane NetworkAttachmentDefinition: Global annotations and commonAnnotations can be added and are correctly merged" {
+    template \
+        --set-json 'interfaces=[{"type": "multus"}]' \
+        --set 'global.annotations.foo=bar' \
+        --set 'commonAnnotations.baz=baa'
+    assert_query_equal '.metadata.annotations.foo' "bar"
+    assert_query_equal '.metadata.annotations.baz' "baa"
+}
+
+@test "Control Plane NetworkAttachmentDefinition: Recommended labels are set" {
+    template --set-json 'interfaces=[{"type": "multus"}]'
+    assert_query_equal '.metadata.labels."app.kubernetes.io/name"' "xrd-control-plane"
+    assert_query_equal '.metadata.labels."app.kubernetes.io/instance"' "release-name"
+    assert_query_equal '.metadata.labels."app.kubernetes.io/managed-by"' "Helm"
+    assert_query '.metadata.labels | has("app.kubernetes.io/version")'
+    assert_query '.metadata.labels | has("helm.sh/chart")'
+}
+
+@test "Control Plane NetworkAttachmentDefinition: Global labels and commonLabels can be added and are correctly merged" {
+    template \
+        --set-json 'interfaces=[{"type": "multus"}]' \
+        --set 'global.labels.foo=bar' \
+        --set 'commonLabels.baz=baa'
+    assert_query_equal '.metadata.labels.foo' "bar"
+    assert_query_equal '.metadata.labels.baz' "baa"
+}
+
+@test "Control Plane NetworkAttachmentDefinition: Check default config" {
+    template --set-json 'interfaces=[{"type": "multus"}]'
+    assert_multiline_query_equal '.spec.config' \
+        "{\n  \"cniVersion\": \"0.3.1\",\n  \"plugins\": [\n    null\n  ]\n}"
+}
+
+@test "Control Plane NetworkAttachmentDefinition: Config can be set for MGMT interfaces" {
+    template --set-json 'mgmtInterfaces=[{"type": "multus", "config": {"foo": "bar"}}]'
+    assert_multiline_query_equal '.spec.config'\
+        "{\n  \"cniVersion\": \"0.3.1\",\n  \"plugins\": [\n    {\n      \"foo\": \"bar\"\n    }\n  ]\n}"
+}
+
+@test "Control Plane NetworkAttachmentDefinition: Config can be set for interfaces" {
+    template --set-json 'interfaces=[{"type": "multus", "config": {"foo": "bar"}}]'
+    assert_multiline_query_equal '.spec.config' \
+        "{\n  \"cniVersion\": \"0.3.1\",\n  \"plugins\": [\n    {\n      \"foo\": \"bar\"\n    }\n  ]\n}"
+}
+
+@test "Control Plane NetworkAttachmentDefinition: No interfaces" {
+    template_failure
+}
+
+@test "Control Plane NetworkAttachmentDefinition: No custom resource created for defaultCNI" {
+    template_failure --set-json 'interfaces=[{"type": "defaultCni"}]'
+}
+
+@test "Control Plane NetworkAttachmentDefinition: error if multiple defaultCNI requested" {
```

Comment:
This is not really a NAD test - the "modules are manifests" model doesn't really work here, and really we want a module to be a more informal grouping of "things in values.yaml that are related" (e.g., in this case interfaces).

I don't think it's worth changing anything - and would suggest just keeping it as is (any change would be a fairly significant change from the established pattern, and I don't want to block this PR which adds significant value)